backend/roles/role-managers/active-mixplay-users.js

```diff
@@ -0,0 +1,142 @@
+    let currentTime = date.getTime();
+
+    let existingUserIndex = activeMixplayUsers.findIndex((obj => obj.userId === user.user_id));
+
+    // If user exists, update their time and stop.
+    if (existingUserIndex !== -1) {
+        logger.debug(user.username + " is still active in mixplay. Updating their time.");
+        activeMixplayUsers[existingUserIndex].time = currentTime;
+        return;
+    }
+
+    // Else, we're going to push the new user to the active array.
+    logger.debug(user.username + " has become active on mixplay. Adding them to active mixplay list.");
+    user = {
+        userId: user.userID,
+        username: user.username,
+        time: currentTime
+    };
+    activeMixplayUsers.push(user);
+}
+
+function removeLeavingUser(username) {
+    for (let userIndex in activeMixplayUsers) {
+        if (activeMixplayUsers[userIndex] != null) {
+            let user = activeMixplayUsers[userIndex];
+            if (user.username === username) {
+                logger.debug(user.username + " has left mixplay. Removing from active mixplay list.");
+                activeMixplayUsers.splice(userIndex, 1);
+            }
+        }
+    }
+}
+
+function clearInactiveUsers() {
+    logger.debug("Clearing inactive people from active mixplay users list.");
+    let date = new Date;
+    let currentTime = date.getTime();
+    let expiredTime = currentTime - inactiveTimer;
+    for (let userIndex in activeMixplayUsers) {
+        if (activeMixplayUsers[userIndex] != null) {
+            let user = activeMixplayUsers[userIndex];
+            if (user.time <= expiredTime) {
+                logger.debug(user.username + " has gone inactive on mixplay. Removing them from active mixplay list.");
+                activeMixplayUsers.splice(userIndex, 1);
+            }
+        }
+    }
+}
+
+function getActiveMixplayUsers() {
+    return activeMixplayUsers;
+}
+
+function cycleActiveMixplayUsers() {
+    let mixplayConnected = Mixplay.mixplayIsConnected;
+    if (!mixplayConnected) {
+        return;
+    }
+
+    logger.info("Starting Active Mixplay Users Loop");
+
+    // Just in case
+    clearInterval(cycleActiveTimer);
+
+    // We have permission to start up the loop now. Let's do this...
+    cycleActiveTimer = setInterval(function() {
+        clearInactiveUsers();
+    }, 60000);
+
+    return;
+}
+
+ipcMain.on("setActiveMixplayUsers", function(event, value) {
+    logger.debug('Changing active mixplay user enabled to: ' + value);
+    if (value === false) {
+        logger.debug('Stopping active mixplay user timeout cycle.');
+        clearInterval(cycleActiveTimer);
+    }
+    activeUserListStatus = settings.getActiveMixplayUserList() === false ? false : true;
+});
+
+ipcMain.on("setActiveMixplayUserTimeout", function(event, value) {
+    logger.debug('Changing active mixplay user timeout to: ' + value);
+
+    // Make sure we have a valid value, then set it.
+    value = parseInt(inactiveTimer);
+    if (isNaN(value)) {
```

Comment:
I would just change this to `value === NaN` since you're doing `parseInt` before the `isNaN` call anyway and `parseInt` returns the value of `NaN` if value isnt a num. `isNaN` is just a convenience method that will automatically parse a string into a number and check if its valid, you dont have to parseInt first. Since you are manually parsing first tho, its unnecessary.

Reply:
It seems like our current linter rules dont like value === NaN for whatever reason.

Reply:
In that case I would call isNaN like before but just move the parseInt call after it, instead of having it before.